private/testeth/network.go

```diff
@@ -0,0 +1,190 @@
+// Copyright (C) 2021 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package testeth
+
+import (
+	"context"
+	"errors"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	"github.com/ethereum/go-ethereum/cmd/utils"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/eth"
+	"github.com/ethereum/go-ethereum/eth/downloader"
+	"github.com/ethereum/go-ethereum/eth/ethconfig"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/node"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/zeebo/errs"
+)
+
+// Network - test Ethererum network.
+type Network struct {
+	ethereum  *eth.Ethereum
+	stack     *node.Node
+	keystore  *keystore.KeyStore
+	developer accounts.Account
+}
+
+// NewNetwork creates new test Ethereum network with PoS and inmemory DBs.
+func NewNetwork() (*Network, error) {
+	config := node.DefaultConfig
+	config.Name = "testeth"
+	config.DataDir = ""
+	config.HTTPHost = "127.0.0.1"
+	config.HTTPPort = 0
+	config.HTTPModules = append(config.HTTPModules, "eth")
+	config.P2P.MaxPeers = 0
+	config.P2P.ListenAddr = ""
+	config.P2P.NoDial = true
+	config.P2P.NoDiscovery = true
+	config.P2P.DiscoveryV5 = false
+
+	stack, err := node.New(&config)
+	if err != nil {
+		return nil, err
+	}
+
+	// setup keystore backend
+	ks := keystore.NewKeyStore(stack.KeyStoreDir(), keystore.LightScryptN, keystore.LightScryptP)
```

Comment:
Just curious: why is this `keystore.NewKeystore` used instead of just generating private keys / addresses with :

```go
key, err := ecdsa.GenerateKey(secp256k1.S256(), rand.Reader)
```